### PR TITLE
Headless/tahsin/brand/seeder&testing

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **Jira** task here for reference.
- [x] Bug fix
- [ ] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [x] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [x] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

**Comment by ghost** [comment]
Jira Link : [https://pm23.atlassian.net/browse/BC-512](url)

File: packages/headless/src/database/mongodb/order/index.ts [hunk 1]
```diff
@@ -28,10 +29,19 @@ export class OrderDatabase implements IOrderDatabase {
...
-      return orderList;
+    if (orderList.length > 0){
+      if(sortType === 'ascending'){
+        return orderList.sort((a,b) => a[sortField] - b[sortField]);
+      }else if(sortType === 'descending'){
+        return orderList.sort((a,b) => b[sortField] - a[sortField]);
+      }else{
+        const sortField: string = "orderedDate";
+        return orderList.sort((a,b) => b[sortField] - a[sortField]);
+      }
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/order/index.ts]: this is not good coding. sorting data by javascript is not good idea. only use sorting using js where necessary and there's no other way.
but here mongodb has both sort and pagination support. you just need to know about that.

File: packages/headless/src/entity/order.ts [hunk 2]
```diff
@@ -163,9 +163,16 @@ export class OrderResponseEntity {
...
   paypalRedirectUrl?: string;
 }
-// export class OrderListWithoutUserId {
 
-// }
+//sorting query entity
+export enum SortTypesDto {
+  ascending = 'ascending',
+  descending = 'descending'
+}
```
Comment (Anjarul-Islam) [inline packages/headless/src/entity/order.ts]: use short form like 'asc', 'desc'

File: packages/headless/src/entity/order.ts [hunk 3]
```diff
@@ -163,9 +163,16 @@ export class OrderResponseEntity {
...
-// export class OrderListWithoutUserId {
 
-// }
+//sorting query entity
+export enum SortTypesDto {
+  asc = 'asc',
+  desc = 'desc'
+}
+export class OrderSortQuery {
+  sortField?: string;
```
Comment (Anjarul-Islam) [inline packages/headless/src/entity/order.ts]: specify some fields, on which you want to provide sorting. don't take anything from front-end.

File: packages/headless/src/entity/order.ts [hunk 4]
```diff
@@ -163,9 +164,29 @@ export class OrderResponseEntity {
...
+  userId = 'userId',
+  billingAddress = 'billingAddress',
+  orderedDate = 'orderedDate',
+  totalCost = 'totalCost',
+  shippingCost = 'shippingCost',
+  productCost = 'productCost',
+  paymentStatus = 'paymentStatus',
+  shippingStatus = 'shippingStatus',
+  orderStatus = 'orderStatus'
+}
```
Comment (Anjarul-Islam) [inline packages/headless/src/entity/order.ts]: among these fields, keep orderDate only. rest of them won't be used in real case,

Comment (Anjarul-Islam) [reply packages/headless/src/entity/order.ts]: also there's no relation of this enum with the model interface! please adjust them too. and when writing anything specific to request/response at first write them in models package and then implement those types in api.

File: packages/headless/src/entity/order.ts [hunk 5]
```diff
@@ -163,9 +164,20 @@ export class OrderResponseEntity {
...
+export enum SortTypesDto {
+  asc = 'asc',
+  desc = 'desc'
+}
+
+export enum SortField{
+  orderedDate = 'orderedDate'
+}
+export class OrderSortQuery {
+  sortField?: string;
```
Comment (Anjarul-Islam) [inline packages/headless/src/entity/order.ts]: here should be sortField: SortField

File: packages/headless/src/database/mongodb/order/index.ts [hunk 6]
```diff
@@ -28,12 +29,18 @@ export class OrderDatabase implements IOrderDatabase {
...
-      return orderList;
+  async getOrderListByUserId(userId: string, sortObj: OrderSortQuery): Promise<OrderEntity[]> {
+    const { sortField, sortType} = sortObj;
+    let sortIndex = -1;
+    const sort = {};
+    if(sortField && sortType){
+      if(sortType === 'asc') sortIndex = 1;
+      sort[sortField] = sortIndex;
+    }else{
+      sort['orderedDate'] = sortIndex;
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/order/index.ts]: why hard-coded property?

Comment (ghost) [reply packages/headless/src/database/mongodb/order/index.ts]: by default orders will be sorted on orderedDate

**Pull request merged**